openTooltipMenu.js

```diff
@@ -0,0 +1,182 @@
+function getFirstLinkText(text) {
+  // find index where movies list starts
+  var movieListStartIndex = text.indexOf('<h2 class="bottom_divider">Movies</h2>');
+  // get all text after that
+  var textAfterMovieListStart = text.substring(movieListStartIndex);
+  // get indices of text between first <li ...>...</li> after movie list start
+  var firstLiOpenAfterListStart = textAfterMovieListStart.indexOf('<li');
+  var firstLiCloseAfterListStart = textAfterMovieListStart.indexOf('</li>');
+  var startTextInd = movieListStartIndex + firstLiOpenAfterListStart;
+  var endTextInd = movieListStartIndex + firstLiCloseAfterListStart;
+  var firstLinkText = text.substring(startTextInd,endTextInd);
+  return firstLinkText;
+}
+
+// gets the match to the regular expression specified by type
+// if type is empty, use optionalRe instead
+function getMatchToRegExp(text,type,optionalRe) {
+  var re;
+  switch (type) {
+    case 'SEARCH_PAGE':
+      re = /(Search Results - Rotten Tomatoes)/;
+      break;
+    case 'NO_RESULTS_FOUND':
+      re = /(no results found)/;
+      break;
+    case 'RATING_VALUE':
+      re = /<span itemprop="ratingValue">(\d*)/g;
+      break;
+    case 'T_METER_SCORE':
+      re = /<span class="tMeterScore">(\d*)/g;
+      break;
+    case 'MOVIE_URL_RESULT_PAGE':
+      re = /<link href="(http:\/\/www.*)" rel="canonical"/;
+      break;
+    case 'MOVIE_URL_SEARCH_PAGE':
+      re = /href="([^<]*)">[^<]*<\/a> <span class="movie_year"> \(\d*\)/;
+      break;
+    case 'TITLE_AND_YEAR_RESULT_PAGE':
+      re = /<title>(.*) - Rotten Tomatoes<\/title>/;
+      break;
+    case 'TITLE_SEARCH_PAGE':
+      re = /href="[^<]*">([^<]*)<\/a> <span class="movie_year"> \(\d*\)/;
+      break;
+    case 'YEAR_SEARCH_PAGE':
+      re = /href="[^<]*">[^<]*<\/a> <span class="movie_year"> \((\d*)\)/;
+      break;
+    default:
+      re = optionalRe;
+  }
+  //console.log(re);
+  var match = re.exec(text);
+  //console.log(match);
+  if (match === null) return undefined
+  return match[1];
+}
+
+function renderTooltipMenuSearchText(str) {
+    $('#tooltipMenu').append('<div>' + str + '</div>');
+    $('#tooltipMenu').css('padding','5px');
+}
+
+function renderTooltipMenuRatingText(str) {
+    $("#tooltipMenu").empty(); // removes any previous divs 
+    if (str.length<=2) {
+      str = 'Rating: ' + str + '%';
+    }
+    $('#tooltipMenu').append('<div>' + str + '</div>');
+    $('#tooltipMenu').css('padding','5px');
+}
+
+function renderTooltipMenuLink(movieUrl,dispTitle) {
+    if (movieUrl === undefined) return
+    var str = dispTitle;
+    if (str === undefined) str = 'Link';// if cannot find title, render 'Link'
+    $('#tooltipMenu').append('<a href=' + movieUrl + ' target=_blank>' + str + '</a>');
+}
+
+function makeSearchString(highlightedText) {
+  // replace spaces with underscores in url
+  var query = highlightedText.split(' ').join('+');
+  return 'https://www.rottentomatoes.com/search/?search=' + query;
```

Comment:
I propose we create a constant at the top of this file to house the base URL for rotten tomatoes.

Then, we can refer to it in both `makeSearchString` and `makeResultString` and flag that it's not intended to change within the scope of our program.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const